Refactored plan edit details and share pages.

briri opened the pull request:
- Refactored the Project details page per #491 
- Refactored the Plan sharing page per #497 
- Consolidated alert/notice/tooltip styling and site's background color to the dmproadmap.scss file so that these settings are more easily changed when branding the site
- Updated alert/notice message colors per #175

xsrust commented on app/policies/plan_policy.rb:
```diff
@@ -56,6 +56,10 @@ def visibility?
   end
 
+  def set_test?
+    @plan.administerable_by?(@user.id)
```
Currently update is set as @plan.editable_by?(@user.id)  

Which one is the correct behavior?
I  can see a case for only owners/co-owners to be able to set visibility, test status, and other plan details

briri replied:
not sure. I'll pose the question to @sjDCC and @stephaniesimms

briri replied:
they both agreed that owners and co-owners should be able to change visibility and set the test flag

xsrust commented on app/views/plans/_plan_title.html.erb:
```diff
@@ -1,8 +1 @@
-    <span><%= _('This is a') %> <strong><%= _('test plan') %></strong>.</span>
-  </div>
-<% end %>
+<h1><%= plan.title %></h1> 
```
Does this still need to be a partial? 
is the intention down the line to move back towards a conditionally different title?

briri replied:
nope. i'll get rid of the world's smallest partial ;)

vyruss commented on app/controllers/plans_controller.rb:
```diff
@@ -38,9 +38,13 @@ def new
-    @plan.data_contact = current_user.email
+    @plan.principal_investigator_email = current_user.email
+    
+    orcid = current_user.identifier_for(IdentifierScheme.find_by(name: 'orcid'))
```
@xsrust I think the identifier_schemes table is convenient because we can tack on descriptions, urls and other values

xsrust replied:
I agree.  I was jusg suggesting that we save the `IdentifierScheme.find_by(name: 'orcid')` within the model for identifier_schemes instead of running a query each time we want to select it.  just like Perms, the content of the record for the orcid identifier scheme will remain constant while the tool is used

briri replied:
agreed @xsrust. I'll open a new ticket for that. A constant would work for this. You need to update the users table, add routes, etc. anyway to add a new identifier scheme, so it requires the site to be bounced.

briri replied:
see ticket #513

vyruss commented on app/views/plans/_edit_details.html.erb:
```diff
@@ -1,95 +1,162 @@
-      <br />
-      <p><% _('Select up to 3 organisations to see their guidance.') %></p>
-      <% #@important_ggs.inspect %>
+      <p><%= _('Select up to 3 organisations to see their guidance.') %></p>
```
I'm not sure about the 3 guidance limit... perhaps we'll need to check that with users.

briri replied:
I didn't actually set the limitation in the code yet. need to ask @sjDCC and @stephaniesimms about it.

xsrust replied:
Durring the call with UX on Wednesday, it was mentioned that 3 was an arbitrary limit due to spacing of the new guidance tabs within plan edit.  If more than 3 are needed, @JEK-III said he would be able to re-design the guidance display to display more, either through a second row of tabs or otherwise.

briri replied:
thanks for the clarification. I'll add some JS logic to prevent the selection of more than 3?

briri commented on app/controllers/plans_controller.rb:
```diff
@@ -152,43 +164,42 @@ def show
-    respond_to :html
-  end
-
+#  def edit
```
yep. will do.

jollopre commented on lib/assets/javascripts/dmproadmap/forms.js:
```diff
@@ -1,39 +1,56 @@
+          .attr('role', (data['code'] === 1 ? 'status' : 'alert'))
+          .attr('class', (data['code'] === 1 ? 'roadmap-info-box' : 'roadmap-alert-box'))
+          .parent().css('visibility', 'visible').fadeIn();
+  });
```
I encourage you to use .then(resolved,rejected) rather than just done, otherwise for any error the user will not be informed at all.

briri replied:
will do

jollopre commented on lib/assets/javascripts/dmproadmap/forms.js:
```diff
@@ -1,39 +1,56 @@
 
 // ---------------------------------------------------------------------------
-function toggleInputError(input, errorMessage){
+function toggleInputError(input, errorMessage, allowBlank = true){
```
I'm not aware of default params working without using a transpiler

briri replied:
thanks. didn't realize it was not reliable

jollopre commented on lib/assets/javascripts/views/plans/index.js:
```diff
@@ -1,23 +1,12 @@
-        e.preventDefault();
-      }
-    });
+    remoteSave("/plans/" + id + "/set_test", 'POST', JSON.stringify(params));
```
remoteSave does not seem like a proper name for this function since is not only "trying to save something on the server" but also changes visibility of elements.

briri replied:
agreed. I went back and forth with 3-4 names before finally giving up. Any suggestions? I didn't want to use something too generic that could cause confusion with other libraries.

jollopre replied:
I would get the body of the done function and would refactor in a new function. Perhaps animateNotification...

briri commented:
- removed unecessary endpoints from plans_controller
- removed _plan_title.html.erb
- added JS logic to prevent user from selecting more than 6 guidance groups (the limit discussed in the UX meeting)
- switched the `remoteSave` ajax wrapper method to `asyncRequest(params, callbacks)`. The `params` value should be a hash of jquery ajax parameters (e.g. `{url: '/my/path', type: 'POST', ...}`) and the `callbacks` should be a hash containing `{success: functionA, failure: functionB}` The method calls the `animateNotification()` function which sets the accessible alert/notice message. See `lib/assets/javascripts/views/plans/index.js` and the `set_test` endpoint on the plans_controller for an example of usage. If all you want to do is set the notification message you can send an empty hash to the function.
- Created a new animateNotification function
- Fixed issue with viewing the read-only version of the plan details
- Updated the 'my dashboard', 'project details' and 'plan share' pages to work with these changes

xsrust commented:
All requested changes made or tickets created so this is looking good to me
